static/js/src/advantage/react/components/Subscriptions/ReBuyExpired/ReBuyExpiredModal.tsx

```diff
@@ -0,0 +1,153 @@
+import React from "react";
+import { UserSubscription } from "advantage/api/types";
+import {
+  Product,
+  Periods,
+  BuyButtonProps,
+} from "advantage/subscribe/react/utils/utils";
+
+type Props = {
+  subscription: UserSubscription;
+  closeModal: () => void;
+};
+
+import PurchaseModal from "PurchaseModal";
+import Summary from "./Summary/Summary";
+import ReBuyExpiredButton from "./ReBuyExpiredButton";
+import usePreview from "advantage/subscribe/react/hooks/usePreview";
+import { useLastPurchaseIds } from "advantage/react/hooks";
+import { selectPurchaseIdsByMarketplaceAndPeriod } from "advantage/react/hooks/useLastPurchaseIds";
+
+const ReBuyExpiredModal = ({ subscription, closeModal }: Props) => {
+  const termsLabel = (
+    <>
+      I agree to the{" "}
+      <a
+        href="/legal/ubuntu-advantage-service-terms"
+        target="_blank"
+        rel="noopener noreferrer"
+      >
+        Ubuntu Pro service terms
+      </a>
+    </>
+  );
+
+  const descriptionLabel = (
+    <>
+      I agree to the{" "}
+      <a
+        href="/legal/ubuntu-pro-description"
+        target="_blank"
+        rel="noopener noreferrer"
+      >
+        Ubuntu Pro description
+      </a>
+    </>
+  );
+
+  const marketingLabel =
+    "I agree to receive information about Canonical's products and services";
+
+  const price: number = subscription.price || 0;
+  const product: Product = {
+    canBeTrialled: false,
+    longId: subscription.listing_id || "",
+    name: subscription.product_name || "",
+    period: Periods.yearly,
+    price: {
+      value: price / subscription.number_of_machines,
+      currency: "USD",
+    },
+    private: false,
+    id: "physical-uai-essential-weekday-yearly", // does not matter
```

Comment:
No because the hook wants a parameter of the type Product for which it's mandatory to provide this id of this specific structure. And I tried to avoid making changes outside of this folder, as I cannot foretell all the potential consequences they might have.

Reply:
The hook should not need to care about more than just the product listing ID and quantity. But then you start going into a rabbit hole of refactoring.